rpmlint/checks/SharedLibraryPolicyCheck.py

```diff
@@ -0,0 +1,311 @@
+            std_lib_package = True
+
+        # ignore libs in a versioned non_std_dir
+        if std_lib_package:
+            for lib in libs.copy():
+                lib_dir = libs_to_dir[lib]
+                if lib_dir.startswith("/opt/kde3"):
+                    continue
+                for lib_part in lib_dir.split('/'):
+                    if len(lib_part) == 0:
+                        continue
+                    if lib_part[-1].isdigit() and not lib_part.endswith("lib64"):
+                        libs.remove(lib)
+                        break
+
+        # Check for non-versioned libs in a std lib package
+        if std_lib_package:
+            for lib in libs.copy():
+                if (not (lib[-1].isdigit() or
+                         self.strongly_versioned_re.search(lib))):
+                    printWarning(pkg, "shlib-unversioned-lib", lib)
+
+        # If this package should be or should be splitted into shlib
+        # package(s)
+        if len(libs) > 0 and len(std_dirs) > 0:
+            # If the package contains a single shlib, name after soname
+            if len(libs) == 1:
+                soname = libs.copy().pop()
+                libname = libname_from_soname(soname)
+                if libname.startswith('lib') and pkg.name != libname and \
+                        pkg.name != libname + "-mini":
+                    printError(pkg, 'shlib-policy-name-error', libname)
+
+            elif not pkg.name[-1:].isdigit():
+                printError(pkg, 'shlib-policy-missing-suffix')
+
+        if (not pkg.name.startswith('lib')) or pkg.name.endswith('-lang'):
+            return
+
+        if not libs:
+            printError(pkg, 'shlib-policy-missing-lib')
+
+        # Verify no non-lib stuff is in the package
+        dirs = set()
+        for f in files:
+            if os.path.isdir(pkg.dirName() + f):
+                dirs.add(f)
+
+        # Verify shared lib policy package doesn't have hard dependency on non-lib packages
+        if std_lib_package:
+            for dep in pkg.requires():
+                if (dep[0].startswith('rpmlib(') or dep[0].startswith('config(')):
+                    continue
+                if (dep[1] & (rpm.RPMSENSE_GREATER | rpm.RPMSENSE_EQUAL)) == rpm.RPMSENSE_EQUAL:
+                    printWarning(pkg, "shlib-fixed-dependency", Pkg.formatRequire(dep[0], dep[1], dep[2]))
+
+        # Verify non-lib stuff does not add dependencies
+        if libs:
+            for dep in pkg_requires.difference(_essential_dependencies):
+                if '.so.' in dep and dep not in libs and dep not in libs_needed:
+                    printError(pkg, 'shlib-policy-excessive-dependency', dep)
+
+        # Check for non-versioned directories beyond sysdirs in package
+        sysdirs = ('/lib', '/lib64', '/usr/lib', '/usr/lib64',
+                   '/usr/share', '/usr/share/licenses',
+                   '/usr/share/doc/packages')
```

Comment:
This should be `/usr/share/doc` instead of `/usr/share/doc/packages`

Reply:
Yep, there is also quite bit of /opt and kde3 prefixes that must go away :)